src/pages/studyView/StudyViewPageStore.ts

```diff
@@ -8912,6 +9087,45 @@ export class StudyViewPageStore
         }
     }
 
+    @observable
+    private genePanelFilters: GenePanelFilter[] = [];
+
+    @action
+    public setGenePanelChartFilters(genePanelFilters: GenePanelFilter[]) {
+        this.genePanelFilters = genePanelFilters;
+    }
+
+    @action
+    public clearGenePanelFilters() {
+        this.genePanelFilters = [];
+    }
+
+    @action.bound
+    public removeGenePanelFilter(filterToRemove: GenePanelFilter) {
+        this.genePanelFilters = this.genePanelFilters.filter(f => {
+            return (
+                f.genePanel != filterToRemove.genePanel &&
+                f.molecularProfileIds != filterToRemove.molecularProfileIds
+            );
+        });
+    }
+
+    private toGenePanelFilter(
```

Comment:
doesn't appear to use chartUniqueKey param.   could probably just a function rather than hiding its reliance on molecularProfiles (and it's being resolved).   i.e. should accept MolecularProfiles[] as argument